pkgs/applications/audio/tauon/default.nix

```diff
@@ -46,26 +46,21 @@ stdenv.mkDerivation (finalAttrs: {
 
   postPatch = ''
     substituteInPlace tauon.py \
-      --replace 'install_mode = False' 'install_mode = True' \
-      --replace 'install_directory = os.path.dirname(os.path.abspath(__file__))' 'install_directory = "${placeholder "out"}/share/tauon"'
+      --replace-fail 'install_mode = False' 'install_mode = True' \
+      --replace-fail 'install_directory = os.path.dirname(os.path.abspath(__file__))' 'install_directory = "${placeholder "out"}/share/tauon"'
 
     substituteInPlace t_modules/t_main.py \
-      --replace 'install_mode = False' 'install_mode = True' \
-      --replace 'libopenmpt.so' '${lib.getLib libopenmpt}/lib/libopenmpt.so' \
-      --replace 'lib/libphazor.so' '../../lib/libphazor.so'
+      --replace-fail 'install_mode = False' 'install_mode = True' \
+      --replace-fail 'libopenmpt.so' '${lib.getLib libopenmpt}/lib/libopenmpt.so'
 
     substituteInPlace t_modules/t_phazor.py \
-      --replace 'lib/libphazor.so' '../../lib/libphazor.so'
+      --replace-fail 'lib/libphazor.so' '../../lib/libphazor.so'
 
-    patchShebangs compile-phazor.sh
-
-    substituteInPlace compile-phazor.sh --replace 'gcc' '${stdenv.cc.targetPrefix}cc'
-
-    substituteInPlace extra/tauonmb.desktop --replace 'Exec=/opt/tauon-music-box/tauonmb.sh' 'Exec=${placeholder "out"}/bin/tauon'
+    substituteInPlace extra/tauonmb.desktop --replace-fail 'Exec=/opt/tauon-music-box/tauonmb.sh' 'Exec=${placeholder "out"}/bin/tauon'
   '';
 
   postBuild = ''
-    ./compile-phazor.sh
+    bash ./compile-phazor.sh
```

Comment:
the patchShebangs that was originally in postPatch should make this unnecessary

Reply:
For some reason, the patchShebangs wasn't working properly, and the script wasn't executing here

This script is only used once while building anyway, so I thought it might just be simpler to call bash directly.

Perhaps it would be good to track down why the patchShebangs wasn't working

Reply:
```
compile-phazor.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/bash"
```

That's not even a valid shebang! That would explain a lot. Strange that it's not a more widespread problem in that case, though.

Reply:
No actually that message should be correct, the message doesn't include the shebang pat of the new line in the output message:
https://github.com/NixOS/nixpkgs/blob/3014c8befc845d1577b0236d4a92a4cdcb27359a/pkgs/build-support/setup-hooks/patch-shebangs.sh#L123-L130

what's weird is that now it's working for me... doing some more testing...

Reply:
I figured, it out, the issue is that `stdenv.cc.targetPrefix` is empty, so the script tries to execute `cc` instead of `gcc`!

Reply:
`patchShebangs` is needed since this particular script hardcodes `#!/bin/bash` which never exists on NixOS. If it was using one of the POSIX-compliant shebangs (`#!/bin/sh` or `#!/usr/bin/env bash`) then it would not be needed.

Reply:
Sorry I was talking about the replacement of gcc in that comment, since that's the actual issue.

Reply:
Nevermind, I see the actual issue! The correct path the shebang to be set to is `/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/bash` as mentioned in the debug log, but it's actually set to `/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2ancc-bash-5.2p32/bin/bash`, which is missing a `g`!